test/js2r-logging-test.el

```diff
@@ -0,0 +1,86 @@
+;;; js2r-formatting-test.el --- Unit tests for js2r-formatting.el  -*- lexical-binding: t; -*-
+
+;; Copyright (C) 2017 Felipe Ochoa
+
+;; Author: Felipe Ochoa <felipe@fov.space>
+;; Keywords: test
+
+;; This program is free software; you can redistribute it and/or modify
+;; it under the terms of the GNU General Public License as published by
+;; the Free Software Foundation, either version 3 of the License, or
+;; (at your option) any later version.
+
+;; This program is distributed in the hope that it will be useful,
+;; but WITHOUT ANY WARRANTY; without even the implied warranty of
+;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;; GNU General Public License for more details.
+
+;; You should have received a copy of the GNU General Public License
+;; along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+;;; Commentary:
+
+;;
+
+;;; Code:
+
+(require 'js2-refactor)
+(require 'buttercup)
+
+(defun js2r--statement-at-point ()
+  "Find the smallest containing statement at point."
+  (js2-node-parent-stmt (js2-node-at-point)))
+
+(describe "finding a suitable log position"
+  (it "should return the preceding line for return statements on their own line"
+    (with-js2-buffer "
+function x() {
+  callFunc();
+  return abc;
+}"
+      (goto-char 40) ;; Just before the a in abc
+      (expect (symbol-at-point) :to-be 'abc)
```

Comment:
This was a sanity check/executable comment to make sure that the otherwise opaque `(goto-char 40)` was at the correct place. I can remove it if you prefer